GenericsHomework/Node.cs

```diff
@@ -0,0 +1,68 @@
+namespace GenericsHomework;
+public class Node<T>
+{
+    // Value of the node
+    public T Value { get; set; }
+
+    // Pointer to the next node
+    public Node<T> Next { get; private set; }
+
+    // Constructor that takes a value
+    public Node(T value)
+    {
+        Value = value;
+       
+        Next = this; // By default, the Next pointer refers back to itself
+     
+      
+    }
+
+    public override string ToString()
+    {
+        return Value!.ToString()!;
+    }
+
+    public void Append(T value)
+    {
+        if (Exists(value))
+        {
+            throw new ArgumentException("Value already exists");
+        }
+        else
+        {
+            Node<T> newNode = new(value)
+            {
+                Next = this.Next // New node points to the current node's next
+            };
+
+            this.Next = newNode; // Update current node's next to point to the new node

```

Comment:
Great job! only simplification
```suggestion
                Next = Next // New node points to the current node's next
            };

            Next = newNode; // Update current node's next to point to the new node

```